src/structures/GuildMember.js

```diff
@@ -283,11 +283,11 @@ class GuildMember extends Base {
    */
   async edit(data, reason) {
     if (data.channel) {
-      data.channel = this.guild.channels.resolve(data.channel);
-      if (!data.channel || data.channel.type !== 'voice') {
+      const voiceChannelID = this.guild.channels.resolveID(data.channel);
+      if (!voiceChannelID) {
```

Comment:
Would probably be nice to keep the type check for cached channels.
Not sure if there is some elegant solution for that though.

Reply:
Added it back in b3ed332. Not sure how elegant it is tho😅